app/assets/javascripts/discourse/app/lib/post-cooked-html-decorators/mentions.gjs

```diff
@@ -55,14 +55,22 @@ export default function (element, context) {
         user.off?.("status-changed", element, _updateUserStatus);
       });
     }
-
-    destroyUserStatusOnMentions();
   };
 }
 
-function _renderUserStatusOnMentions(mentions, user, owner) {
+function _renderUserStatusOnMentions(mentions, user, helper) {
   mentions.forEach((mention) => {
-    updateUserStatusOnMention(owner, mention, user.status);
+    let wrapper = mention.querySelector(".user-status-message-wrapper");
+    if (!wrapper) {
+      wrapper = document.createElement("span");
+      wrapper.classList.add("user-status-message-wrapper");
+      mention.appendChild(wrapper);
+    }
+
+    helper.renderGlimmer(mention, CookedUserStatusMessage, {
```

Comment:
This is cool! I wonder what the performance impact will be when we have 10s/100s of renderGlimmer in a post... will be interesting to see.